### DISCUSSION TITLE
Local songs support

### DISCUSSION THREAD

**Discussion by YajanaRao** [YajanaRao/Serenity#88] [Ideas]
Play local songs from device. 
- [x] Scanning all the folders for songs
- [x] Option to play all songs, add or remove from queue and delete song

**Comment by YajanaRao** [comment]
Planning of using [react-native-fs](https://github.com/itinance/react-native-fs#readme). Also need to work on 

- [x] Getting the [meta data from mp3](https://github.com/venepe/react-native-music-metadata#readme) 
- [ ] Need to look into [Musicbrainz](https://musicbrainz.org/doc/Developer_Resources) for fetching the meta data.

**Reply by Lucyy-cloud** [reply]
The best tool to choose the option of browsing songs and editing metadata based on albums and artists (if any) is called the [Viwizard Amazon Music Converter](https://www.viwizard.com/amazon-music-converter/), which is used to download songs from Amazon Music Prime, Amazon Music Unlimited and Amazon Music HD and convert them to MP3, AAC, WAV, FLAC, M4A and M4B, while preserving the original sound quality and ID3 tag information. In the settings window, you can select the output format and adjust the bit rate, sampling rate, and channel.

**Comment by YajanaRao** [comment]
For getting meta data instead of using one more library, can it be done in [react-native-fs](https://github.com/itinance/react-native-fs/issues/683)

**Comment by YajanaRao** [comment]
- [x] Option to browse songs based on albums and artists if present
- [ ] Option to edit meta data
